### PR TITLE
18085 create service helper

### PR DESCRIPTION
**Pull request by tungtx-2038**
Feature #18072 Search Screen
+ create servicehelper to manage api service 
Redmine:
+[Task#18091: create ServiceHelper](https://edu-redmine.sun-asterisk.vn/issues/18091)

### CODE REVIEW

File: WeatherForcast/WeatherForcast/Data/ServiceHelper.swift [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+    private init(_ currentService: CurrentService, _ fiveDayService: FiveDayService, _ placeService: PlaceService, _ uvService: UVIndexService, _ airService: AirPolutionService) {
+        self.currentService = currentService
+        self.fiveDayService = fiveDayService
+        self.placeService = placeService
+        self.uvService = uvService
+        self.airService = airService
+    }
+
+    static func getInstance(_ currentService: CurrentService, _ fiveDayService: FiveDayService, _ placeService: PlaceService, _ uvService: UVIndexService, _ airService: AirPolutionService) -> ServiceHelper {
```
Comment (dongtg-1485) [inline WeatherForcast/WeatherForcast/Data/ServiceHelper.swift]: brealine nếu tên function quá dài

File: WeatherForcast/WeatherForcast/Data/ServiceHelper.swift [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+final class ServiceHelper {
+
+    private var currentService: CurrentService
+    private let fiveDayService: FiveDayService
+    private let placeService: PlaceService
+    private let uvService: UVIndexService
+    private let airService: AirPolutionService
+
+    private init(_ currentService: CurrentService, _ fiveDayService: FiveDayService, _ placeService: PlaceService, _ uvService: UVIndexService, _ airService: AirPolutionService) {
```
Comment (dongtg-1485) [inline WeatherForcast/WeatherForcast/Data/ServiceHelper.swift]: breakline nếu tên function quá dài

File: WeatherForcast/WeatherForcast/Data/ServiceHelper.swift [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+        self.airService = airService
+    }
+
+    static func getInstance(_ currentService: CurrentService, _ fiveDayService: FiveDayService, _ placeService: PlaceService, _ uvService: UVIndexService, _ airService: AirPolutionService) -> ServiceHelper {
+        return ServiceHelper(currentService, fiveDayService, placeService, uvService, airService)
+    }
+
+
+
+    func getWeather(param: CurrentWeatherParams, onSuccess: @escaping ((_ data: CurrentWeather) -> Void), onFailed: @escaping Failure) {
```
Comment (dongtg-1485) [inline WeatherForcast/WeatherForcast/Data/ServiceHelper.swift]: breakline giữa các param

File: WeatherForcast/WeatherForcast/Extensions/Int+Extension.swift [hunk 4]
```diff
@@ -19,4 +19,15 @@ extension Int {
         let stringDate = dateFormat.string(from: date)
         return stringDate
     }
+
+    func getStringHourFromUnix() -> String {
+        let myDouble = Double(self)
+        let date = Date(timeIntervalSince1970: myDouble)
+        let dateFormat = DateFormatter()
```
Comment (datdt-1935) [inline WeatherForcast/WeatherForcast/Extensions/Int+Extension.swift]:
```suggestion
        let dateFormat = DateFormatter(). then {
```

File: WeatherForcast/WeatherForcast/Data/ServiceHelper.swift [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+        self.uvService = uvService
+        self.airService = airService
+    }
+
+    static func getInstance(_ currentService: CurrentService, _ fiveDayService: FiveDayService,
+                            _ placeService: PlaceService, _ uvService: UVIndexService,
+                            _ airService: AirPolutionService) -> ServiceHelper {
+        return ServiceHelper(currentService, fiveDayService, placeService, uvService, airService)
+    }
+
```
Comment (longvt-1468) [inline WeatherForcast/WeatherForcast/Data/ServiceHelper.swift:38]: remove line

File: WeatherForcast/WeatherForcast/Scenes/DetailScreen/SubViewPageViewController/TemperatureCell/TemperatureTableViewCell.swift [hunk 6]
```diff
@@ -11,7 +11,50 @@ import Reusable
...
+}
+
+extension TemperatureTableViewCell: UICollectionViewDataSource {
+
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return weather.count
+    }
+
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        let cell = temperatureCollection.dequeueReusableCell(for: indexPath, cellType: TemperatureCollectionViewCell.self).then {
```
Comment (longvt-1468) [inline WeatherForcast/WeatherForcast/Scenes/DetailScreen/SubViewPageViewController/TemperatureCell/TemperatureTableViewCell.swift:51]: break line

**Review by dongtg-1485 — APPROVED** [review]

**Review by tuanlvt-1117 — APPROVED** [review]

**Pull request merged**